src/pages/index.js

```diff
@@ -70,31 +141,95 @@ function handleImageClick(name, link) {
   newImagePopup.open({ link, name });
 }
 
-const cardSection = new Section(
-  {
-    items: cardData,
-    renderer: renderCard,
-  },
-  ".cards__list"
-);
-
 function renderCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  );
   const cardElement = card.getView();
-  cardListEl.prepend(cardElement);
+  cardSection.addItem(cardElement);
 }
 
-cardSection.renderItems();
+function handleDeleteClick(card) {
+  newPopupConfirm.open();
+  newPopupConfirm.setSubmitAction(() => {
+    api.deleteCard(card.id).then(() => {
+      card.deleteCard();
+      newPopupConfirm.close();
+    });
+  });
+}
+
+function handleLikeClick(card) {
+  if (!card.isLiked) {
+    api.likeCard(card.id).then(() => {
+      card.updateLikes(true);
+    });
+  } else {
+    api.dislikeCard(card.id).then(() => {
+      card.updateLikes(false);
+    });
+  }
+}
 
 function handleProfileFormSubmit({ name, description }) {
-  userInfo.setUserInfo({ name, description });
-  newProfilePopup.close();
+  newProfilePopup.renderLoading(true);
+  api
+    .updateProfileInfo({ title: name, description })
+    .then(() => {
+      userInfo.setUserInfo({ name, description });
+      editProfileFormValidator.disableButton();
+      newProfilePopup.close();
+    })
+    .catch((error) => {
+      console.error("Error updating profile:", error);
+      editProfileFormValidator.resetValidation();
```

Comment:
`resetValidation` is not needed in `catch`. It's needed only when you open the **profile** modal to clear possible errors from the inputs



**Please, correct it everywhere**

Reply:
Okay I moved them to be inside of the click event for each modal, I hope that is correct based off what you explained.